### ISSUE TITLE
format function

### ISSUE THREAD

**Issue by alimanfoo** [alimanfoo/petl#295]
Add a format() function which is based on convert, but supports the specific case where values should be formatted into a string using the string.format() function. E.g.:

```
format(tbl, 'foo', '{:02d}')
```

Maybe also add support for alternative formatting via the modulus operator, e.g.:

```
interpolate(tbl, 'foo', '%02d')
```

**Comment by alimanfoo** [comment]
Could also imagine adding a field based on formatting using a row.

**Comment by alimanfoo** [comment]
implemented format() and interpolate() in v1.0
